pydantic/utils.py

```diff
@@ -83,6 +83,15 @@ def in_ipython() -> bool:
         return True
 
 
+def deep_update(mapping: Dict[str, Any], updating_mapping: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
KeyType = TypeVar('KeyType')

```suggestion
def deep_update(mapping: Dict[KeyType, Any], updating_mapping: Dict[KeyType, Any]) -> Dict[KeyType, Any]:
```

I don't think the key type here has to be a string, we might use the method elsewhere.